sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -356,6 +356,17 @@ public void writeTo(ByteBuffer buffer) {
     buffer.position(pos + sizeInBytes);
   }
 
+  public void copyTo(UnsafeArrayData toArray) {
+    // Copy header
+    Platform.copyMemory(this.baseObject, this.baseOffset + 8,
+      toArray.baseObject, toArray.baseOffset + 8,
+      Math.min(this.elementOffset - this.baseOffset, toArray.elementOffset - toArray.baseOffset));
+    // Copy values
+    Platform.copyMemory(this.baseObject, this.elementOffset,
+      toArray.baseObject, toArray.elementOffset,
+      Math.min(this.sizeInBytes - this.elementOffset, toArray.sizeInBytes - toArray.elementOffset));
+  }
```

Comment:
This method was needed to support "expanding" UnsafeArrayData object so that an intermediate expandable array is not needed.